webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/OrderedScrambles.java

```diff
@@ -0,0 +1,245 @@
+                        
+            DateTimeZone timezone = DateTimeZone.forID(removeQuotation(venue.getAsJsonObject().get("timezone").toString()));
+            DateTime competitionStartDate = new DateTime(competitionStartString, timezone);
+            
+            boolean hasMultipleRooms = venue.getAsJsonObject().getAsJsonArray("rooms").size()>1;
+
+            for (JsonElement room : venue.getAsJsonObject().getAsJsonArray("rooms")) {
+
+                ArrayList<Integer> dayList = new ArrayList<Integer>();
+                ArrayList<ArrayList<ScrambleRequest>> scrambleRequestListByDay = new ArrayList<ArrayList<ScrambleRequest>>();
+                
+                String roomName = removeQuotation(room.getAsJsonObject().get("name").toString());
+                                
+                for (JsonElement activity : room.getAsJsonObject().getAsJsonArray("activities")) {
+                    String activityCode = removeQuotation(activity.getAsJsonObject().get("activityCode").toString());
+
+                    String[] activityList = activityCode.split("-");
+                    String eventId = activityList[0];
+
+                    if (!eventId.equals(wcifIgnorableKey)) {
+
+                        DateTime activityStartTime = new DateTime(removeQuotation(activity.getAsJsonObject().get("startTime").toString()), timezone);
+
+                        int activityDay = Days.daysBetween(competitionStartDate.withTimeAtStartOfDay(), activityStartTime.withTimeAtStartOfDay()).getDays()+1;
+                        
+                        if (!dayList.contains(activityDay)) {
+                            dayList.add(activityDay);
+                            scrambleRequestListByDay.add(new ArrayList<ScrambleRequest>());
+                        }
+
+                        int round = -1;
+                        int group = -1;
+                        int attempt = -1;
+                        
+                        // This part assumes every round, group and attempt is labeled with an integer from competitionJson
+                        for (String item : activityList) {
+                            if (item.charAt(0) == 'r') {
+                                round = Integer.parseInt(item.substring(1));
+                            } else if (item.charAt(0) == 'g') {
+                                group = Integer.parseInt(item.substring(1));
+                            } else if (item.charAt(0) == 'a') {
+                                attempt = Integer.parseInt(item.substring(1));
+                            }
+                        }
+
+                        ScrambleRequest[] scrambleRequests = GSON.fromJson(sheet, ScrambleRequest[].class);
+                        
+                        // first, we add all requests whose events equals what we need
+                        ArrayList<ScrambleRequest> scrambleRequestTemp = new ArrayList<ScrambleRequest>();
```

Comment:
nit: It might be cleaner/easier to first build a `HashMap` of `String eventId` -> `ScrambleRequest[]` and then do a lookup into it at this point.

Reply:
Since this takes writing an iteration anyways and it doesn't change too much on computation, unless you have strong feeling about this, I'd rather not to use for this (moved to the new WCIFHandler class).